src/Services/EctsScrapper/ectsScrapper.service.ts

```diff
@@ -0,0 +1,33 @@
+import { getAllDegreeURLs } from './getAlldegreeURL.service';
+import { getAllSubjectsDegreeURLs } from './getAllSubjectsURLs.service';
+import { scrappedEctsSubjects } from './utils/scrappedEctsSubjects';
+import checkSubjects from '../../utils/scrappers/ectsScrapper/checkSubjects';
+import { returnScraperError } from '@/utils/errorScraper';
+
+export const getAllUrls = async () => {
+    {
+        try {
+            const allDegreeURLs = await getAllDegreeURLs();
+            const allSubjectsURLs = await getAllSubjectsDegreeURLs(
+                allDegreeURLs
+            );
+
+            const specialCases = allSubjectsURLs
+                .filter((el, _) => checkSubjects(el.name))
+                .map((el, _) => {
+                    return { url: el.url, degree: el.degree };
+                });
```

Comment:
I think it could work like that but it's Your decision whether it is readable:

```suggestion
                .map((el, _) => ({ url: el.url, degree: el.degree }));
```